### DISCUSSION TITLE
Media Element custom controls

### DISCUSSION THREAD

**Discussion by ne0rrmatrix** [CommunityToolkit/Maui#1114] [Q&A]
I am looking to add a custom control to the media element controls bar. Is that something that can be done? I have figured out how to add full screen and restore to normal for media element on windows and android. It is not ready for adding to toolkit as I think the methods I used are not appropriate and I lack the skill to add it toolkit unfortunately. 

I have it working as a img button in the top right corner that lets you click on it for full screen on windows/android but I would like to add it directly to the video controls bar to the right of existing controls. If I can't do that  I would like to have the full screen control fade out like the control bar does. 

I have tried a few different methods all of which have depended on touch controls working. I had forgotten that I could not really use gesture controls as they are blocked by media element atm. If anyone has any suggestions I would appreciate it.

**Comment by ne0rrmatrix** [comment]
I have finished implementing device specific controls for media element in my project. Really happy with how I implemented it. I have it as device specific code that is called as function within a content view that has media element in it. On any page you can just use this code as example:
```
<media:MediaControl x:Name="mediaElement" Source="{Binding Url}" />
```
This will let you have full screen controls like on youtube with a little icon in top right that you can click on to make full screen or not.
The function you use is either FullScreen() or RestoreScreen() and it works for both windows and android. It uses platform specific code and outside of MediaControl class and platform code it needs the developer to do nothing. 

I am going to work on adding it to toolkit as an project. Once I have it fully fleshed out I will open a discussion to see if there is interest from the community to add full screen to the controls element of media Element. I also need to know what the requirements would be for implementation as this would be my first contribution to any open source project.

In another week or two I will have it added to a forked version of media element and I will seek input as to how to implement it properly before starting a discussion. I plan on doing this as a learning project. Learning dotnet and Maui has been fun and I am enjoying it immensely.

**Reply by pictos** [reply]
@ne0rrmatrix that's amazing! I would say that this is the desired feature. Please read our proposal flow (you can find it on our Read me file), you can start a discussion ASAP, with the API spec

**Comment by CliffAgius** [comment]
Closed and answered
